DeratControl.Domain/Entities/Perimeter.cs

```diff
@@ -13,13 +13,26 @@ public enum PerimeterType
     }
     public class Perimeter : EntityBase<int>
     {
-        public int FacilityId { get; set; } 
-        public virtual Facility Facility { get; set; }
-        public PerimeterType PerimeterType { get; set; }
+        public int FacilityId { get; protected set; }
+        public virtual Facility Facility { get; protected set; }
+        public PerimeterType PerimeterType { get; protected set; }
         public virtual IEnumerable<Point> TrapPoints { get; protected set; }
         private Perimeter()
         {
             this.TrapPoints = new List<Point>();
         }
+        public Perimeter(int facilityId, Facility facility, PerimeterType perimeterType, int createdBy)
```

Comment:
does it make any sense to pass facilityId separately if you already pass a Facility object?

Reply:
resolved requested changes except make FindById asynchronous because of my teammate already doing this